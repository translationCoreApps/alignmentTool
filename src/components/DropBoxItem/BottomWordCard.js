import React, {Component} from 'react';
import PropTypes from 'prop-types';
import { DragSource } from 'react-dnd';
import ItemTypes from '../ItemTypes';

const internalStyle = {
  display: 'flex',
  width: '100%',
  borderLeft: '5px solid #44C6FF',
  padding: '10px',
  textAlign: 'center',
  backgroundColor: '#FFFFFF',
  boxShadow: "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
  cursor: 'move'
};

class BottomWordCard extends Component {
  render() {
    const { word, style, isDragging, connectDragSource } = this.props;
    const opacity = isDragging ? 0.4 : 1;

    return connectDragSource(
      <span style={{ ...internalStyle, ...style, opacity }}>
        {word}
      </span>
    );
  }
}

BottomWordCard.propTypes = {
  word: PropTypes.string.isRequired,
  occurrence: PropTypes.number.isRequired,
  occurrences: PropTypes.number.isRequired,
  alignmentIndex: PropTypes.number,
  connectDragSource: PropTypes.func.isRequired,
  isDragging: PropTypes.bool.isRequired,
  style: PropTypes.object
};

const DragBottomWordCardAction = {
  beginDrag(props) {
    // Return the data describing the dragged item
    const item = {
      word: props.word,
      occurrence: props.occurrence,
      occurrences: props.occurrences,
      alignmentIndex: props.alignmentIndex
    };
    return item;
  }
};

const collect = (connect, monitor) => {
  return {
    connectDragSource: connect.dragSource(),
    connectDragPreview: connect.dragPreview(),
    isDragging: monitor.isDragging()
  };
};

export default DragSource(
  ItemTypes.BOTTOM_WORD,
  DragBottomWordCardAction,
  collect
)(BottomWordCard);
