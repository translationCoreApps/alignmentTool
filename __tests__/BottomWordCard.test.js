import React from 'react';
import BottomWordCard from '../src/components/DropBoxItem/BottomWordCard';

test('BottomWordCard renders', () => {
    const component = (
        <BottomWordCard />
    );
    expect(component).toMatchSnapshot();

    // TODO: exercise UI
});