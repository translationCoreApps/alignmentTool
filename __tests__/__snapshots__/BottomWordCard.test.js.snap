// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomWordCard renders 1`] = `<DragSource(BottomWordCard) />`;
